### ISSUE TITLE
2.6 Do not display player in Help> Unit help Table, List of players not to show (1of 4)

### ISSUE THREAD

**Issue by TheDog-GH** [triplea-game/triplea#12646]

**Comment by gittarsem** [comment]
Is this resolved or assigned to anyone

**Comment by TheDog-GH** [comment]
@gittarsem 
No not resolved/assigned.

**Comment by gittarsem** [comment]
@TheDog-GH Kindly assign this task to me

**Comment by TheDog-GH** [comment]
@Cernelius 
Although I got your post as an email, it is not above.  So I copy it here.

> Arguably also not showing units which are like those players even if belonging to a proper player, or even totally absent from the game.
>
> The Unit Help table of World At War comes to mind, having a bunch of units which are not in the game at all, but that would be better done by actually excluding units which a player can never receive (according to the XML).

**Comment by TheDog-GH** [comment]
@Cernelius 
Never played the game, had a quick look at the units folder, its tidy in the sense there is no units in the \units folder to be shared with all factions.  So is it not the case, just delete the units in each faction that cannot be bought?

**Comment by gittarsem** [comment]
hi @TheDog-GH ,
I have implemented support for players_not_shown_in_unit_help_table from map.properties, and the players listed there are correctly excluded from the Unit Help table.
While testing with players_not_shown_in_unit_help_table=Neutral_True, I noticed that the "Neutral Units" section is still displayed. This section comes from the engine's NullPlayer (Neutral), not the Neutral_True player.
Should the NullPlayer also be hidden in this case, or is the property intended to hide only the explicitly listed players?

**Comment by TheDog-GH** [comment]
Yes the option for map-makers to have the choice to NOT list NullPlayer would be good.  Its clutter the players dont need to see.

**Comment by gittarsem** [comment]
Hi @TheDog-GH, this PR addresses issue #12646. The feature has been implemented, tested, and passes the project checks. I'd appreciate your review and approval if everything looks good. Thank you!

**Comment by TheDog-GH** [comment]
@frigoref Hi, please could you review, thanks.

**Comment by frigoref** [comment]
@TheDog-GH which PR? This is issue #12646

**Comment by TheDog-GH** [comment]
Yes this one please.

**Comment by Cernelius** [comment]
> As per Cernelius suggestion below this should be in map.properties

Hey, can you please give me a link to wherever I said what you mention? I remember that I said that this rather belongs to map.properties than the game-file (XML), but I'm not sure that I remember everything I said, and now I'm thinking this is probably better somewhere else in the main (map) folder. Thank you.

**Comment by TheDog-GH** [comment]
@Cernelius 
Here is the discussion
https://github.com/triplea-game/triplea/issues/12636

**Comment by Cernelius** [comment]
> [@Cernelius](https://github.com/Cernelius) Here is the discussion [#12636](https://github.com/triplea-game/triplea/issues/12636)

Thank you.

Warning: nit picking - Regarding
>As per Cernelius suggestion below this should be in map.properties

looks like that what I said is actually that "it should be handled at a skin level like the map.property "dont_draw_units"", not that it should specifically be in that file ("map.properties") (it could, not should). By "at a skin level", I meant anywhere inside the "map" folder but outside the "games" folder. Nevertheless, your summary of my opinion is good because then you proceed proposing about having it actually inside "map.properties" and looks like I did not object about it, so I practically sort of subscribed to it.

Let's say I changed my mind and now think it would be better having it in an other file akin to (and in the same folder as) the "map.properties" file (but I'm not strongly against having it in "map.properties").

**Comment by TheDog-GH** [comment]
I think let the Devs choose where the options goes.

map.properties or xml (could be part of the <playerList>)

Im easy, just having the options will reduce clutter for the player will be great.
